code/modules/mob/living/simple_animal/abnormality/he/red_shoes.dm

```diff
@@ -0,0 +1,448 @@
+			continue
+		potentialmarked += L
+	for(var/i=numbermarked, i>=1, i--)
+		var/mob/living/Y = pick(potentialmarked)
+		if(faction_check_mob(Y, FALSE) || Y.z != z || Y.stat == DEAD)
+			continue
+		if(Y in marked)
+			continue
+		marked+=Y
+		Apply_Desire(Y)
+	datum_reference.qliphoth_change(2)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/red_shoes/proc/Assimilate(mob/living/carbon/user)
+	if(!(status_flags & GODMODE))
+		return
+	if(possessee)
+		return
+	var/list/search_area = user.contents//get items from the possessee
+	possessee = user
+	var/mob/living/carbon/human/H = user
+	if(ishuman(H) && (H.sanity_lost))//also assimilates fortitude panics
+		for(var/obj/item/clothing/suit/armor/ego_gear/EQ in search_area)//copies all resistances from worn E.G.O
+			for(var/damtype in damage_coeff)
+				if(damtype == BRUTE)
+					continue
+				damage_coeff[damtype] -= (EQ.armor[damtype] / 100)
+			break//check only the first
+		user.forceMove(src)
+		playsound(src, 'sound/abnormalities/redshoes/RedShoes_Activate.ogg', 50, 1)
+		name = user.name//later reminder to drop weapons and apply sanguine desire's inhands as an overlay here
+		appearance = user.appearance
+		gender = user.gender
+		desc = "[user.name] appears to be grinning from ear to ear. Does [p_they()] normally wear shoes like those?"
+		maxHealth += (user.maxHealth * 4.5)
+		revive(full_heal = TRUE, admin_revive = FALSE)
+		add_overlay(mutable_appearance('icons/mob/clothing/feet.dmi', "red_shoes", -ABOVE_MOB_LAYER))
+		add_overlay(mutable_appearance('icons/mob/inhands/weapons/ego_righthand.dmi', "sanguine", -ABOVE_MOB_LAYER))
+		BreachEffect(user)
+		return TRUE
+
+
+/mob/living/simple_animal/hostile/abnormality/red_shoes/proc/UnPossess(mob/living/carbon/human/user)//called on death() and BreachEffect()
+	var/datum/status_effect/panicked_type/red_possess/S = user.has_status_effect(/datum/status_effect/panicked_type/red_possess)
+	if(S)
+		user.cut_overlay(mutable_appearance('icons/effects/effects.dmi', "murder", -ABOVE_MOB_LAYER))
+		QDEL_NULL(user.ai_controller)
+		user.ai_controller = /datum/ai_controller/insane/murder
+		user.InitializeAIController()
+		user.apply_status_effect(/datum/status_effect/panicked_type/murder)
+
+//BreachEffect and combat
+/mob/living/simple_animal/hostile/abnormality/red_shoes/BreachEffect(mob/living/carbon/human/user)
+	if(!(status_flags & GODMODE))
+		return
+	soundloop.stop()
+	for(var/mob/living/carbon/human/H in GLOB.mob_living_list)//stops possessing people, prevents runtimes. Panicked players are ghosted so use mob_living_list
+		UnPossess(H)
+	if(!possessee)
+		name = "Red Shoe"
+		desc = "The Red Shoes’s bloody enameled leather glistens in the light."
+		icon_state = "redshoes_breach"
+		icon_living = "redshoes_breach"
+		damage_coeff = list(BRUTE = 1, RED_DAMAGE = 0.5, WHITE_DAMAGE = 1.5, BLACK_DAMAGE = 1, PALE_DAMAGE = 1.5)
+		var /mob/living/simple_animal/hostile/red_shoe/S = new(get_turf(src))
+		S.Initialize()
```

Comment:
```suggestion
```
This should be automatically done by the system and does not need to be called manually.